restli-common/src/main/java/com/linkedin/restli/common/validation/RestLiDataSchemaDataValidator.java

```diff
@@ -66,6 +71,24 @@ public ValidationResult validateOutput(RecordTemplate dataTemplate)
     return super.validateOutputAgainstSchema(dataTemplate, _validatingSchema);
   }
 
+  /**
+   * Validator to use to validate the output.
+   * The validator is instantiated in the constructor, so directly returns that if input is equal to _validatingSchema.
+   * @param validatingSchema schema to validate against
+   * @return validator
+   */
+  @Override
+  protected Validator getValidatorForOutputEntityValidation(DataSchema validatingSchema) {
+    if (_validatingSchema == validatingSchema)
```

Comment:
if the strict pointer equality check is on purpose, please add an inline doc on why it's the case

Reply:
+1 on Ivan's suggestion